### DISCUSSION TITLE
Any current workarounds for 2fa (two factor authentication) for CI pipeline?

### DISCUSSION THREAD

**Discussion by rankinit** [fastlane/fastlane#17201] [Q&A]
It sounds like long term there will be a better solution for handling the 2fa issues when your account has 2fa (two factor authentication) turned on.
https://github.com/fastlane/fastlane/discussions/16812#discussioncomment-35677

However in the meantime is there any workarounds for 2fa while this is being built?  Our CICD pipeline works great except when the FASTLANE_SESSION expires and we need to manually renew it (every 2 -3 weeks).

Thanks for any help!

**Comment by rankinit** [comment]
No workarounds?

**Comment by StefaniOSApps** [comment]
make a App Store account without 2FA

**Reply by max-ott** [reply]
This is no longer possible. In the meantime use an (old) Apple ID without 2FA or wait for Josh's changes to `match` and similar to be able to simply use an API key.

**Reply by rankinit** [reply]
Thanks for input and will persist with 2fa for the time being until the new changes are available.
